File: lclang/src/main/java/lclang/libs/lang/BoolClass.java
package lclang.libs.lang;

public class BoolClass extends LibraryClass {
    public static String name = "bool";
    public static final BoolClass TRUE = new BoolClass(true);
    public static final BoolClass FALSE = new BoolClass(false);

    public boolean bool;
    public BoolClass() {}
    public BoolClass(boolean bool) {
        this.bool = bool;
    }

    @Override
    public boolean equals(Object other) {
        if(other instanceof BoolClass)
            return ((BoolClass) other).bool == bool;

        return false;
    }

    @Override
    public String toString() {
        return String.valueOf(bool);
    }
}

File: lclang/src/main/kotlin/lclang/libs/lang/BoolClass.kt
package lclang.libs.lang

import lclang.LCClass

class BoolClass(): LCClass("bool") {
    var bool: Boolean = false
    constructor(bool: Boolean): this() {
        this.bool = bool
    }

    override fun equals(other: Any?): Boolean {
        if(other is BoolClass)
            return other.bool == bool

        return false
    }

    override fun toString(): String = bool.toString()
    override fun hashCode(): Int = bool.hashCode()

    companion object {
        val TRUE = BoolClass(true)
        val FALSE = BoolClass()
    }
}